### PR TITLE
Prediction/eval card aggregation

### PR DESCRIPTION
**Pull request by sgsmob**
Introduce a function `evalcast::aggregate_cards()` to unpack a list of prediction or evaluation cards into a single unnested data frame.

Completes the implementation described in #221 although the new function is used in any existing function (TODO in a follow-on PR).

### CODE REVIEW

File: R-packages/evalcast/R/manage_cards.R [hunk 1]
```diff
@@ -66,11 +66,65 @@ intersect_locations <- function(cards) {
...
+#'     `card$forecast_distribution` across each `card` in `list_of_cards`.  That is to say, if each
+#'     card has 7 rows in its `forecast_distribution` and there are 3 such cards in
+#'     `list_of_cards`, the output will have 21 rows.
+#'   \item The columns of the output correspond to:
+#'     \itemize{
+#'       \item Columns of `card$forecast_distribution`
+#'       \item Columns of `card`
+#'       \item Attributes of `card`
+#'     }
+aggregate_cards <- function(list_of_cards) {
```
Comment (dajmcdon) [inline R-packages/evalcast/R/manage_cards.R:85]: I think you need to export this function.

File: R-packages/evalcast/R/manage_cards.R [hunk 2]
```diff
@@ -66,11 +66,65 @@ intersect_locations <- function(cards) {
...
+#' This is a generic method for dispatching to specific calls for evaluation and prediction cards.
+#' @param card Evaluation or prediction card.
+#' @return See `aggregate_cards`.
+unpack_single_card <- function(card){
+  UseMethod("unpack_single_card", card)
+}
 
+#' Unpack a single prediction card into an unnested tibble
+unpack_single_card.prediction_card <- function(card) {
+  card_attr = attributes(card)
```
Comment (dajmcdon) [inline R-packages/evalcast/R/manage_cards.R:100]: Just for lintr credit, replace `=` with `<-`

File: R-packages/evalcast/R/manage_cards.R [hunk 3]
```diff
@@ -66,11 +66,65 @@ intersect_locations <- function(cards) {
...
+}
 
-
-
-
-
-
+#' Unpack a single evaluation card into an unnested tibble
+unpack_single_card.evaluation_card <- function(card) {
+  card_attr = attributes(card)
```
Comment (dajmcdon) [inline R-packages/evalcast/R/manage_cards.R:117]: See above lintr suggestion.

File: R-packages/evalcast/tests/testthat/test-manage-cards.R [hunk 4]
```diff
@@ -0,0 +1,187 @@
+library(tibble)
```
Comment (dajmcdon) [inline R-packages/evalcast/tests/testthat/test-manage-cards.R:1]: `get_predictions()` can produce a list of tibbles. The most obvious way would be by setting `ahead=1:4`. I just want to ensure that this works as well.

**Review by dajmcdon — CHANGES_REQUESTED** [review]
Mostly minor plus one question for the tests.

File: R-packages/evalcast/R/manage_cards.R (hunk 1 above)
Comment (sgsmob) [reply R-packages/evalcast/R/manage_cards.R:86]: Done

File: R-packages/evalcast/R/manage_cards.R (hunk 2 above)
Comment (sgsmob) [reply R-packages/evalcast/R/manage_cards.R]: Done

File: R-packages/evalcast/tests/testthat/test-manage-cards.R (hunk 4 above)
Comment (sgsmob) [reply R-packages/evalcast/tests/testthat/test-manage-cards.R:1]: The tests do have lists of tibbles so I am not sure what you are asking about here.

File: R-packages/evalcast/R/manage_cards.R (hunk 3 above)
Comment (sgsmob) [reply R-packages/evalcast/R/manage_cards.R]: Done

File: R-packages/evalcast/R/manage_cards.R (hunk 1 above)
Comment (dajmcdon) [reply R-packages/evalcast/R/manage_cards.R:86]: Sorry, I should have been more explicit. It needs `#' @export` on its own line above `aggregate_cards` or else the function won't be available to the package user.

**Review by dajmcdon — APPROVED** [review]

**Pull request merged**